Fixed Annotations HitTesting & Added annotation anchor

iakov-kaiumov opened the pull request:
This PR implements several fixes and enchantments:
1. Clickable SwiftUI Annotations! #4 
2. Added `anchor` parameter to the `ViewMapAnnotation` similar to the default SwiftUI Map approach.
3. Fixed annotations centering problem #14. Thanks to @rderimay (I took solution from #25)

P.S.
@pauljohanneskraft 
The problem with your hittesting solution in the hit-testing branch was that the SwiftUI annotation view was not constrained to the underlying UIView. Therefore, when you moved the map, the SwiftUI view's position did not change. You may see in the picture below.
<img width="408" alt="Снимок экрана 2023-03-04 в 21 59 45" src="https://user-images.githubusercontent.com/29120192/222954705-6ff2c4e1-a244-4c0c-8182-4de8e7014798.png">

pauljohanneskraft commented:
Thank you for your contribution. I will open a similar pull request in the coming days, since I disagree with the addition of directions into this library (you could easily add this right now, so it doesn't need to be integrated in here). Also it seems that you dropped watchOS 6.0 support - however, this library tries to keep supporting all platforms starting at watchOS 6, macOS 10.15, iOS 13.0 and tvOS 13.0.

iakov-kaiumov commented:
@pauljohanneskraft
Hello! Thank you for your response. I now understand that integrating directions was pretty useless. I reverted this commit. Also, I'm sorry that I accidentally dropped watchOS 6.0 support while merging changes from #25. I reverted the changes and returned watchOS 6.0 support.

I would be happy to open new PR with these changes if you don't mind.
